code/game/objects/items/devices/radio/radio.dm

```diff
@@ -258,10 +261,28 @@ GLOBAL_LIST_INIT(default_medbay_channels, list(
 	if(jammed)
 		message = Gibberish(message, 100)
 	var/list/message_pieces = message_to_multilingual(message)
-	Broadcast_Message(connection, A,
-						0, "*garbled automated announcement*", src,
-						message_pieces, from, "Automated Announcement", from, "synthesized voice",
-						4, 0, zlevel, connection.frequency, follow_target = follow_target)
+
+		// Make us a message datum!
+	var/datum/tcomms_message/tcm = new
+	tcm.connection = connection
+	tcm.sender = A
+	tcm.radio = src
+	tcm.sender_name = from
+	tcm.message_pieces = message_pieces
+	tcm.sender_job = "Automated Announcement"
+	tcm.vname = "synthesized voice"
+	tcm.data = SIGNALTYPE_AINOTRACK
+	// Datum radios dont have a location (obviously
+	if(loc && loc.z)
+		tcm.source_level = loc.z // For anyone that reads this: This used to pull from a LIST from the CONFIG DATUM. WHYYYYYYYYY!!!!!!!! -aa
+	else
+		tcm.source_level = 1 // Assume Z1 if we dont have an actual Z level available to us.
+	tcm.freq = connection.frequency
+	tcm.follow_target = follow_target
+
+	// Now put that through the stuff
+	for(var/obj/machinery/tcomms/core/C in GLOB.tcomms_machines)
+		C.handle_message(tcm)
 	qdel(A)
```

Comment:
This is why I noticed the problem with de-ref'ing; `tcm` here is holding a reference to the announcer.

Now, all mobs are hard deleted, so this doesn't actually matter, but...

Reply:
Since been fixed